Data Migration For New Structure

prescottprue opened the pull request:
* Converts `needs` -> `requests`, `requests_public`, `requests_contact`
* Copies `needs/$needId/history` -> `requests/$requestId/actions`
* Converts `userProfiles` -> `users`, `users_privileged`, `users_public`

Commit:
Start new version migration

Commit:
Add copying of needs history

Commit:
Add g/l

thebitguru committed:
Fixing some behavior based on the old data structure.

thebitguru commented on bin/newVersionMigration.js:
```diff
@@ -0,0 +1,228 @@
+        d: {
+          firstName: profileSnap.get('d.firstName'),
+          displayName: profileSnap.get('d.displayName'),
+          // TODO: we have to call the scramble location on coordinates because of the current
```
Any chance you can add this part?

thebitguru reviewed (changes requested):
I made some quick changes to account for the source data, but now I am running into the following error.

```
Converting to new data format...
Error converting userProfiles -> users: 3 INVALID_ARGUMENT: maximum 500 writes allowed per request
```

Besides that, we need to scramble location as indicated in my comment.

I also created a new project for testing the migration and gave you access to it.  It has the current production data except for the subcollection issue that I emailed about.

Commit:
Merge branch 'next' into new-version-migration

prescottprue commented:
To copy the subcollections in Fireadmin you can pass the path of the subcollection and it should copy. I will look into why it wasn't copying with subcolleciton to begin with

As for the error in the migration, it looks like it hit the batch of 500, so I'll change the migration to work with less in each batch.

Commit:
Add generalLocationName through reverse geocoding

Commit:
Fix Geopoints. Handle formatted address not being in geocode result.

thebitguru committed:
fix(newVersionMigration): fix the request_actions doc creation.
fix(newVersionMigration): all userProfiles always have a location and numerous other adjustments to match the latest data model.
fix(newVersionMigration): add delay to conversion to address the Google Geocoding API rate limit.
feat(deleteNewCollections): add temporary script to delete collections taht the migration script creates.
feat(testQuery): add simple test query script.

thebitguru committed:
Merge branch 'next' into new-version-migration
